### PR TITLE
[gettext] Add "autopoint" tool to gettext[core]

### PR DESCRIPTION
**Pull request by dg0yt**
"autopoint" is needed for building some packages, when running `autoreconf`. It is included in the "tools" feature of gettext, but this feature has a longer buildtime, and only a tiny part of this feature is actually needed.

- #### What does your PR fix?  
  ~This PR adds an explicit "autopoint" feature to gettext. Other than the slow build of "tools", it uses a custom cmake script to create the output.~
  ~In addition, it updates the libidn2 port to use this feature, resolving #17057.~
  ~It is an alternative to adding to adding autopoint to CI provisioning. Related discussion is in #17227.~
  This PR adds a lightweight (cmake-based) installation of `autopoint` to the core feature of gettext. This makes it available for the autoconf step of all ports that use gettext at runtime.

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  all, -/-

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  yes

### CODE REVIEW

**Comment by JackBoosY** [comment]
cc @Neumann-A

File: ports/gettext/install-autopoint.cmake [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+set(datadir [[${datarootdir}]])
+set(exec_prefix [[${prefix}]])
+set(PATH_SEPARATOR ":")
+set(RELOCATABLE "yes")
+
+file(STRINGS ${SOURCE_PATH}/gettext-tools/configure
+  VERSIONS_FROM_CONFIGURE
+  REGEX "^ *(ARCHIVE_VERSION|VERSION)=.*$"
+)
+foreach(VERSIONS_FROM_CONFIGURE IN LISTS VERSIONS_FROM_CONFIGURE)
```
Comment (dg0yt) [inline ports/gettext/install-autopoint.cmake:15]: Oops, wrong loop variable identifier. Same on next line.

File: ports/gettext/install-autopoint.cmake [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+endif()
+file(COPY ${SOURCE_PATH}/gettext-tools/misc/archive.dir.tar DESTINATION ${WORKING_DIR})
+vcpkg_execute_required_process(
+  COMMAND gzip -f archive.dir.tar
+  WORKING_DIRECTORY ${WORKING_DIR}
+  LOGNAME gzip-${TARGET_TRIPLET}
+)
+
+# copyright file
+configure_file(${SOURCE_PATH}/COPYING ${WORKING_DIR}/copyright COPYONLY)
```
Comment (dg0yt) [inline ports/gettext/install-autopoint.cmake:40]: ${SOURCE_PATH}/gettext-tools/COPYING

Comment (dg0yt) [reply ports/gettext/install-autopoint.cmake:41]: No, gettext-tools/COPYING just refers to COPYING in parent dir.

**Comment by autoantwort** [comment]
> /azp run

```
$ git commit --allow-empty -m "Trigger Build"
$ git push
```

**Comment by autoantwort** [comment]
It generates a `autopoint` executable, but the `gtkdocize` executable is still missing.

**Comment by dg0yt** [comment]
I took libidn2 (merge conflict) out of this PR now.

#17406 indicated that port fontconfig really needs `autopoint`. Maybe this is true for other ports as well. Vcpkg CI happens to provide it on all systems, but vcpkg users might lack it. In particular, it is a separate package on Debian/Ubuntu.

I leave this open as a draft for now, until it is more clear if this is needed and desired.

**Comment by dg0yt** [comment]
Refreshed the patch. I think it might still be useful, to remove the explicit need to install system autopoint (Ubuntu) or brew gettext (macOS).

**Comment by JackBoosY** [comment]
Please ping me if this PR is ready for review.

**Comment by dg0yt** [comment]
@JackBoosY It is ready for review. 
One question for discussion is whether either gettext tools should be moved to `tools/gettext` (which is automatically added to PATH), or if `vcpkg-cmake-config.cmake` should generally adjust the PATH to include `tools/gettext/bin`. This would remove the need to explicitly add PATH setup from consuming portfiles (hunspell, fontconfig, ...). Unfortunately, my question for file installation layout (#17607) did lead to a satisfying answer.

**Comment by dg0yt** [comment]
Another user fell over autopoint: #19226

File: ports/gettext/portfile.cmake [hunk 3]
```diff
@@ -1,13 +1,7 @@
-if(VCPKG_TARGET_IS_LINUX)
-    if (NOT EXISTS "/usr/include/libintl.h")
-        message(FATAL_ERROR "When targeting Linux, `libintl.h` is expected to come from the C Runtime Library (glibc). "
-                            "Please use \"sudo apt-get install libc-dev\" or the equivalent to install development files."
-        )
-    endif()
-    if(NOT "tools" IN_LIST FEATURES)
-        set(VCPKG_POLICY_EMPTY_PACKAGE enabled)
-        return()
-    endif()
```
Comment (JackBoosY) [inline ports/gettext/portfile.cmake:10]: How long to build this tool in vcpkg?

Comment (dg0yt) [reply ports/gettext/portfile.cmake:10]: Feaure "tools"? Now with https://github.com/microsoft/vcpkg/pull/18159 merged: 

"CI build time for gettext[tools]:x64-windows is now down to 15 ... 18 min"
(https://github.com/microsoft/vcpkg/pull/18159#issuecomment-865265120)

In contrast, feature "autopoint" here takes 0 s.

**Review by JackBoosY — APPROVED** [review]

**Comment by BillyONeal** [comment]
Does it make sense to make this be a feature in default-features and leave the previous "core" behavior alone?

**Comment by dg0yt** [comment]
> Does it make sense to make this be a feature in default-features and leave the previous "core" behavior alone?

Yes. In April, this PR was initially submitted for non-default feature. I can change that back. 

I could also add a host dependency for this feature in port fontconfig, as a validation. However, it probably won't make a difference in CI without additional changes: This port's tools are not installed to a dir which is automatically added to the path. Due to #17607, it is unclear how to improve this in general.

**Comment by BillyONeal** [comment]
> > Does it make sense to make this be a feature in default-features and leave the previous "core" behavior alone?
> 
> Yes. In April, this PR was initially submitted for non-default feature. I can change that back.

I was more making sure it had been considered than asking for a change.  I know nothing about gettext other than hating that it is a tiny library that consumes more than an hour to build for no reasonable reason.

**Comment by dg0yt** [comment]
> What's the difference in build time before and after this change for customers who only cared about [core] before?

Less than one second? That's the whole point of this PR. This feature is entirely implemented in CMake, in the context of the portfile, instead of using gettext's build system. 

For simplicity, the PR uses gzip from MSYS2 for compressing data. gettext defaults to xz. For more consistency, I could probably change this PR to xz, too, using CMake 3.18's `file(ARCHIVE_EXTRACT/ARCHIVE_CREATE ...)`.

Vcpkg and gettext is toxic combination. Someone with MSVC should capture the cached configuration test results and add them to the port to cut down the time spent in configuration.

**Pull request merged**